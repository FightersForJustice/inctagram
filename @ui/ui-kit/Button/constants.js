export const BUTTON_COLORS = Object.freeze({
  PRIMARY: 'Primary',
  BASIC: 'Basic',
  OUTLINED: 'Outlined',
  GHOST: 'Ghost',
})

export const BUTTON_VARIATIONS = Object.freeze({
  FULL_WIDTH: 'FullWidth',
  AUTO_HEIGHT: 'AutoHeight',
  DEFAULT: 'Default',
})
