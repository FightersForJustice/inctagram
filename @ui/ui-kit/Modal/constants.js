export const CLOSE_MODAL_TEST_ID = 'closeModal'
