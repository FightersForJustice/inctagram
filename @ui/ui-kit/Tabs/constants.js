export const TAB_COLORS = Object.freeze({
  PRIMARY: 'Primary',
  GHOST: 'Ghost',
})
export const profileSettingsData = [
  { value: 'general', label: 'General information' },
  { value: 'Devices', label: 'Devices' },
  { value: 'AccountManagment', label: 'Account Managment' },
  { value: 'Payments', label: 'MyPayments' },
]
