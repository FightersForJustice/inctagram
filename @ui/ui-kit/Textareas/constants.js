export const TEXTAEREA_COLORS = Object.freeze({
  DEFAULT: 'Default',
  ACTIVE: 'Active',
  ERROR: 'Error',
})
