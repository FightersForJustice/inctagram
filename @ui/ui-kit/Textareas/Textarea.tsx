import React, { ChangeEvent, FocusEvent } from 'react'
import classNames from 'classnames'
import styles from './Textarea.module.scss'
import { TEXTAEREA_COLORS } from './constants'

export type TextAreaProps = {
  hasError?: boolean
  disabled?: boolean
  hovered?: boolean
  errorMessage?: string
  placeholder?: string
  color?: (typeof TEXTAEREA_COLORS)[keyof typeof TEXTAEREA_COLORS]
} & React.TextareaHTMLAttributes<HTMLTextAreaElement> // Inherit all textarea HTML attributes

export const TextArea: React.FC<TextAreaProps> = ({
  defaultValue,
  hasError = false,
  disabled = false,
  hovered = false,
  errorMessage = 'Error text',
  placeholder = 'Textarea',
  color = TEXTAEREA_COLORS.DEFAULT,
  ...restProps // Spread the remaining HTML attributes
}) => {
  const textareaClasses = classNames(styles.textarea, {
    [styles.Default]: color === TEXTAEREA_COLORS.DEFAULT,
    [styles.Active]: color === TEXTAEREA_COLORS.ACTIVE,
    [styles.Error]: color === TEXTAEREA_COLORS.ERROR,
  })

  return (
    <div>
      <p className={styles.title}>Textarea</p>
      <textarea
        className={textareaClasses}
        defaultValue={defaultValue}
        disabled={disabled}
        placeholder={placeholder}
        {...restProps} // Spread the remaining HTML attributes
      />
      {hasError && <p className={styles.error_message}>{errorMessage}</p>}
    </div>
  )
}
