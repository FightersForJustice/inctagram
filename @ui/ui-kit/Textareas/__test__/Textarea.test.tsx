import React from 'react'
import { render, fireEvent, cleanup, screen } from '@testing-library/react'
import { TextArea } from '../Textarea'
import styles from './Textarea.module.scss'
import { TEXTAEREA_COLORS } from '../constants'

const setupTextArea = (props = {}) => {
  const defaultProps = {
    onChange: jest.fn(),
    onFocus: jest.fn(),
    ...props,
  }

  render(<TextArea {...defaultProps} />)

  return {
    textarea: screen.getByRole('textbox'),
    ...defaultProps,
  }
}

const value = 'type sth here...'

describe('TextArea snapshots \t', () => {
  afterEach(cleanup)

  const textProps = ['', 'Some text', 'Custom placeholder']
  const colorProps = [TEXTAEREA_COLORS.DEFAULT, TEXTAEREA_COLORS.ACTIVE, TEXTAEREA_COLORS.ERROR]
  const booleanProps = [true, false]

  textProps.forEach((text) => {
    colorProps.forEach((color) => {
      booleanProps.forEach((hasError) => {
        booleanProps.forEach((disabled) => {
          booleanProps.forEach((hovered) => {
            it(`text: "${text}", color: ${color}, hasError: ${hasError}, disabled: ${disabled}, hovered: ${hovered}`, () => {
              const props = {
                value: text,
                color: color,
                hasError: hasError,
                disabled: disabled,
                hovered: hovered,
              }
              const { asFragment } = render(<TextArea {...props} />)

              expect(asFragment()).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})

describe('Textarea', () => {
  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
  })

  it('does not respond to events when disabled', () => {
    const { textarea, onChange } = setupTextArea({ disabled: true })

    fireEvent.change(textarea, { target: { value } })
    fireEvent.focus(textarea)

    expect(onChange).not.toHaveBeenCalled()
  })

  it('renders correctly', () => {
    const { textarea } = setupTextArea()

    expect(textarea).toBeInTheDocument()
  })

  it('changes value and calls onChange', () => {
    const { textarea, onChange } = setupTextArea()
    fireEvent.change(textarea, { target: { value } })

    expect(textarea).toHaveValue(value)
    expect(onChange).toHaveBeenCalledTimes(1)
  })

  it('renders label', () => {
    const label = 'Test Label'
    const { textarea } = setupTextArea({ label })
    const labelElement = screen.getByText(label)

    expect(labelElement).toBeInTheDocument()
    expect(labelElement).toBeVisible()
  })

  it('displays error message when hasError is true', () => {
    const errorMessage = 'Test Error Message'
    setupTextArea({ hasError: true, errorMessage })

    const error = screen.getByText(errorMessage)
    expect(error).toBeInTheDocument()
  })

  it('is disabled when disabled prop is true', () => {
    const { textarea } = setupTextArea({ disabled: true })

    expect(textarea).toBeDisabled()
  })
  it('is not disabled when disabled prop is missing', () => {
    const { textarea } = setupTextArea()

    expect(textarea).not.toBeDisabled()
  })
  it('is not disabled when disabled prop is false', () => {
    const { textarea } = setupTextArea({ disabled: false })

    expect(textarea).not.toBeDisabled()
  })

  it('renders placeholder', () => {
    const { textarea } = setupTextArea()

    expect(textarea).toHaveAttribute('placeholder')
  })
  it('checks placeholder value', () => {
    render(<TextArea />)
    const textarea = screen.getByRole('textbox')

    expect(textarea).toHaveAttribute('placeholder')
    expect(textarea).toMatchSnapshot()
  })

  it('has default styles when color prop is not provided', () => {
    const { textarea } = setupTextArea()

    expect(textarea).toHaveClass(styles.Default)
  })
  it('has Active styles when Active color prop is provided', () => {
    const { textarea } = setupTextArea({ color: 'Active' })

    expect(textarea).toHaveClass(styles.Active)
  })
  it('applies Active class when hovered', () => {
    setupTextArea({ hovered: true })

    const textareaElement = screen.getByRole('textbox')
    expect(textareaElement).toHaveClass('Active')
  })
  it('does not apply Active class when not hovered', () => {
    setupTextArea({ hovered: false })

    const textareaElement = screen.getByRole('textbox')
    expect(textareaElement).not.toHaveClass('Active')
  })
  it('has Error styles when Error color prop is provided', () => {
    const { textarea } = setupTextArea({ color: 'Error' })

    expect(textarea).toHaveClass(styles.Error)
  })
})
