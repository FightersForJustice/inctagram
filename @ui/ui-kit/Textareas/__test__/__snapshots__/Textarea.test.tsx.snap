// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea snapshots 	 text: "", color: Active, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Active, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Active, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Active, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Active, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Active, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Active, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Active, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Default, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Default, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Default, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      disabled=""
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Default, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      disabled=""
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Default, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Default, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Default, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      disabled=""
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Default, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      disabled=""
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Error, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Error, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Error, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      disabled=""
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Error, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      disabled=""
      placeholder="Textarea"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Error, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Error, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Error, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      disabled=""
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "", color: Error, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      disabled=""
      placeholder="Textarea"
    />
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Active, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Active, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Active, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Active, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Active, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Active, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Active, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Active, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Default, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Default, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Default, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Default, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Default, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Default, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Default, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Default, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Error, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Error, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Error, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Error, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Error, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Error, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Error, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Custom placeholder", color: Error, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      disabled=""
      placeholder="Textarea"
    >
      Custom placeholder
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Active, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Active, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Active, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Active, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Active, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Active, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Active, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Active, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Default, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Default, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Default, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Default, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Default, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Default, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Default, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Default, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Default Active"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Error, hasError: false, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Error, hasError: false, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Error, hasError: false, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Error, hasError: false, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Error, hasError: true, disabled: false, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Error, hasError: true, disabled: false, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Error, hasError: true, disabled: true, hovered: false 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Error"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`TextArea snapshots 	 text: "Some text", color: Error, hasError: true, disabled: true, hovered: true 1`] = `
<DocumentFragment>
  <div>
    <p
      class="title"
    />
    <textarea
      class="textarea Active Error"
      disabled=""
      placeholder="Textarea"
    >
      Some text
    </textarea>
    <p
      class="error_message"
    >
      Error text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Textarea checks placeholder value 1`] = `
<textarea
  class="textarea Default"
  placeholder="Textarea"
/>
`;
